packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  ResourceSnapshotsReportModel,
+  ResourceSnapshotsReportOperationModel,
+  ResourceSnapshotsReportResultCode,
+  ResourceSnapshotsReportStatus,
+} from '@coveord/platform-client';
+import {
+  ReportViewerOperationName,
+  ReportViewerResourceReportModel,
+} from './reportViewer/reportViewerDataModels';
+
+export class SnapshotReporter {
+  public constructor(public readonly report: ResourceSnapshotsReportModel) {}
+
+  public hasChangedResources() {
+    const totalUnchanges =
+      this.getOperationTypeTotalCount('resourcesUnchanged');
+    return totalUnchanges !== this.report.resourcesProcessed;
+  }
+
+  public getOperationTypeTotalCount(
+    type: keyof ResourceSnapshotsReportOperationModel
+  ) {
+    const count = Object.values(this.report.resourceOperations).reduce(
+      (count: number, current: ResourceSnapshotsReportOperationModel) =>
+        count + current[type],
+      0
+    );
+
+    return count;
+  }
+
+  public isSuccessReport(): boolean {
+    const {status, resultCode} = this.report;
+    return (
+      status === ResourceSnapshotsReportStatus.Completed &&
+      resultCode === ResourceSnapshotsReportResultCode.Success
+    );
+  }
+
+  public getChangedResources(
+    operationsToDisplay: ReportViewerOperationName[]
+  ): ReportViewerResourceReportModel[] {
+    type resourceEntries = [string, ResourceSnapshotsReportOperationModel];
```

Comment:
I would not declare a `type` in a scope (i.e function or class).
This is a hurdle to reusability IMHO.

Reply:
I understand. However, this type will only be used in this function and I did not want to make it public outside that scope.
It simply to avoid copying the type twice

Reply:
nvm. I just  saw your comment below :P

Reply:
> I understand. However, this type will only be used in this function and I did not want to make it public outside that scope.
> It simply to avoid copying the type twice

If you don't export it, it would only be available in the file. If you need to scope a type more narrowly than a file, that's kinda sus imho.